### PR TITLE
Prepare for version 2.4

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Removed a bunch of deprecated code, update related tests. The most consequential change was that the code to unphase data sets phased with old style phasing was moved to the `fix_phase` method. The tests related to this also required new test files to be written out with the old style of phasing (because it can no longer be done on the fly in the test)

I also:
- Removed a skip from a UVCal test related to initializing from UVData objects which exposed a small bug that I fixed.
- Made the `UVData.from_file` method inherit the `UVData.read` docstring using docstring_parser.
- Cleaned up a couple minor docs issues that were generating warnings when building the docs with sphinx.

## Motivation and Context

This should close out our version 2.4 milestone.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [x] Version change
- [ ] Build or continuous integration change

Breaking change checklist:
- [x] I have updated the docstrings associated with my change using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have updated the tutorial to reflect my changes (if appropriate).
- [ ] My change includes backwards compatibility and deprecation warnings (if possible).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

Version change checklist:
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md) to put all the unreleased changes under the new version (leaving the unreleased section empty).
- [ ] I have noted any dependency changes since the last version and will update the conda package build accordingly.

### CODE REVIEW

File: pyuvdata/uvflag/uvflag.py [hunk 1]
```diff
@@ -1054,6 +993,8 @@ def check(self, check_extra=True, run_check_acceptability=True):
         elif self.type == "antenna":
             if self.antenna_numbers is not None:
                 if not all(ant in self.antenna_numbers for ant in self.ant_array):
+                    print(self.antenna_numbers)
+                    print(np.unique(self.ant_array))
```
Comment (mkolopanis) [inline pyuvdata/uvflag/uvflag.py:997]: Do we want these prints here? Maybe just state in the error message which antennas don't belong to both?

**Review by mkolopanis — COMMENTED** [review]
Looking good so far. There are a few place in testing mostly where rogue print statements live. I have read the TODO's around them but still wonder if we want to keep the prints around.

There's one additional print I'm pretty sure you didn't mean to leave in.

File: pyuvdata/uvdata/tests/test_uvdata.py [hunk 2]
```diff
@@ -1398,29 +1243,18 @@ def test_old_phasing(future_shapes):
...
-    # the tolerances here are empirical -- based on what was seen in the
-    # external phasing test. See the phasing memo in docs/references for
-    # details.
-    assert np.allclose(uvd1_drift.uvw_array, uvd2_drift.uvw_array, atol=2e-2)
-    assert np.allclose(uvd1_drift_antpos.uvw_array, uvd2_drift_antpos.uvw_array)
+    assert np.allclose(uvd1_drift.uvw_array, uvd2_drift.uvw_array)
+    # the tolerances here are empirical -- this just makes sure they don't get worse.
+    # TODO: more investigation here needed!
+    print(np.max(np.abs(np.real(uvd1_drift.data_array - uvd2_drift.data_array))))
+    print(np.max(np.abs(np.imag(uvd1_drift.data_array - uvd2_drift.data_array))))
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply pyuvdata/uvdata/tests/test_uvdata.py:1256]: yeah, they're cruft, I'll take them out.

File: pyuvdata/uvflag/uvflag.py (hunk 1 above)
Comment (bhazelton) [reply pyuvdata/uvflag/uvflag.py:997]: Oops, I meant to take those out. But I think you're right, putting them in the error message is even better.

**Comment by bhazelton** [comment]
Thanks @mkolopanis! I removed the print statements, fixed some coverage issues and added some things I'd missed before to the changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,6 +28,27 @@ the `erfa` or `novas` astrometry libraries.
 - Error when setting `UVBeam.freq_interp_kind` to an integer.
 - Error when reading `mwa_corr_fits` files from the new MWAX correlator
 
+### Removed
+- The `phase_uvw` and `unphase_uvw` utility methods associated with old style phasing.
+- The `with_conjugates` option to the `get_baseline_redundancies` utility method in
+favor of the `include_conjugates` option.
+- Support for the the `interpolation_function` attribute on UVBeams.
```
Comment (steven-murray) [inline CHANGELOG.md:35]: typo: two "the"s

**Review by steven-murray — COMMENTED** [review]
Looks good -- just the one typo that I found. I love seeing so much red!

File: CHANGELOG.md (hunk 3 above)
Comment (bhazelton) [reply CHANGELOG.md]: fixed! Thanks for catching that.

**Review by mkolopanis — APPROVED** [review]

**Pull request merged**